pyuoi/linear_model/base.py

```diff
@@ -303,16 +303,14 @@ def fit(self, X, y, stratify=None, verbose=False):
                 else:
                     # reset the coef
                     if hasattr(self.selection_lm, 'coef_'):
-                        self.selection_lm.coef_ = np.zeros_like(
-                            self.selection_lm.coef_,
-                            dtype=X_test.dtype,
-                            order='F')
+                        self.selection_lm.coef_ *= 0.
                     # predict by fitting an empty design matrix
-                    self.selection_lm.fit(np.zeros_like(X_test), y_test)
+                    self.selection_lm.fit(
```

Comment:
The logistic L1 solver breaks with an input with zero variance. This is a hack that works around that problem, but it would probably be better to figure something else out.

Reply:
^ if there is no intecept

Reply:
Ultimately, the solution to this might be to create a `_fit_intercept` function which calculates the maximum likelihood intercept value given a feature set. We would then set the coefficients and intercept by hand and pass that through the `score_predictions` function.